### ISSUE TITLE
[bitnami/kafka] doc request: clarify that using config overrides all environment vars

### ISSUE THREAD

**Issue by JoshKCarroll** [bitnami/charts#3958]
**Which chart**:
bitnami/kafka, not new but running 11.8.6

**Describe the bug**
TL;DR: Currently setting a value for `config:` will clobber most other chart configuration, and this wasn't obvious from the docs as a new naive user. Which caused a significant headache that could have been alleviated with a small documentation update.

It appears that the intention of the current chart config is to either use the provided values properties (which set a variety of `KAFKA_CFG_*` environment variables), or fully specify a config file either via a string value for `config:` or an `existingConfigMap`.

As a naive new user to the chart, we expected that we could set a partial configuration for any kafka broker configs that aren't explicitly listed in the chart values using the `config:` option and it would be merged with environment variables. However, what we realized after much consternation is that when any value is set for `config:`, the bitnami kafka docker image seems to ignore `KAFKA_CFG_*` environment variables.

After discovering this and re-reading the docs, I can see that the initial interpretation was perhaps a bit naive, and seems the preferred way is adding `extraEnvVars: {KAFKA_CFG_EXTRA_VAR: ...}`, which is reasonable. I just wish this was more clear on first reading.

This is especially important for settings that may be dynamic or differ across brokers, such as `zookeeper.connect` and `advertised.listeners`.

**To Reproduce**

Here's an example of a naive install that shows how the chart breaks unexpectedly when I set a minimal config.

```sh
$ helm install bitnami/kafka --set config="queued.max.requests: 16" --version 11.8.6 --namespace test-kafka --name test-kafka
NAME:   test-kafka
LAST DEPLOYED: Thu Oct  8 15:17:58 2020
NAMESPACE: test-kafka
STATUS: DEPLOYED

RESOURCES:
==> v1/ConfigMap
NAME                      DATA  AGE
test-kafka-configuration  1     1s
test-kafka-scripts        1     1s

==> v1/Pod(related)
NAME                    READY  STATUS             RESTARTS  AGE
test-kafka-0            0/1    ContainerCreating  0         1s
test-kafka-zookeeper-0  0/1    ContainerCreating  0         1s

==> v1/Service
NAME                           TYPE       CLUSTER-IP     EXTERNAL-IP  PORT(S)                     AGE
test-kafka                     ClusterIP  100.71.195.12  <none>       9092/TCP                    1s
test-kafka-headless            ClusterIP  None           <none>       9092/TCP,9093/TCP           1s
test-kafka-zookeeper           ClusterIP  100.67.135.21  <none>       2181/TCP,2888/TCP,3888/TCP  1s
test-kafka-zookeeper-headless  ClusterIP  None           <none>       2181/TCP,2888/TCP,3888/TCP  1s

==> v1/ServiceAccount
NAME        SECRETS  AGE
test-kafka  1        1s

==> v1/StatefulSet
NAME                  READY  AGE
test-kafka            0/1    1s
test-kafka-zookeeper  0/1    1s


NOTES:


** Please be patient while the chart is being deployed **

Kafka can be accessed by consumers via port 9092 on the following DNS name from within your cluster:

    test-kafka.test-kafka.svc.cluster.local

Each Kafka broker can be accessed by producers via port 9092 on the following DNS name(s) from within your cluster:

    test-kafka-0.test-kafka-headless.test-kafka.svc.cluster.local:9092

To create a pod that you can use as a Kafka client run the following commands:

    kubectl run test-kafka-client --restart='Never' --image docker.io/bitnami/kafka:2.6.0-debian-10-r30 --namespace test-kafka --command -- sleep infinity
    kubectl exec --tty -i test-kafka-client --namespace test-kafka -- bash

    PRODUCER:
        kafka-console-producer.sh \
            
            --broker-list test-kafka-0.test-kafka-headless.test-kafka.svc.cluster.local:9092 \
            --topic test

    CONSUMER:
        kafka-console-consumer.sh \
            
            --bootstrap-server test-kafka.test-kafka.svc.cluster.local:9092 \
            --topic test \
            --from-beginning

$ kubectl get pods -n test-kafka
NAME                     READY   STATUS             RESTARTS   AGE
test-kafka-0             0/1     CrashLoopBackOff   4          3m2s
test-kafka-zookeeper-0   1/1     Running            0          3m2s
$ kubectl logs test-kafka-0 -n test-kafka
 19:19:54.82 
 19:19:54.82 Welcome to the Bitnami kafka container
 19:19:54.82 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-kafka
 19:19:54.82 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-kafka/issues
 19:19:54.82 
 19:19:54.83 INFO  ==> ** Starting Kafka setup **
 19:19:54.87 WARN  ==> You set the environment variable ALLOW_PLAINTEXT_LISTENER=yes. For safety reasons, do not use this flag in a production environment.
 19:19:54.88 INFO  ==> Initializing Kafka...
 19:19:54.88 INFO  ==> ** Kafka setup finished! **

 19:19:54.89 INFO  ==> ** Starting Kafka **
[2020-10-08 19:19:55,663] INFO Registered kafka:type=kafka.Log4jController MBean (kafka.utils.Log4jControllerRegistration$)
[2020-10-08 19:19:56,109] ERROR Exiting Kafka due to fatal exception (kafka.Kafka$)
org.apache.kafka.common.config.ConfigException: Missing required configuration "zookeeper.connect" which has no default value.
	at org.apache.kafka.common.config.ConfigDef.parseValue(ConfigDef.java:478)
	at org.apache.kafka.common.config.ConfigDef.parse(ConfigDef.java:468)
	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:108)
	at org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:142)
	at kafka.server.KafkaConfig.<init>(KafkaConfig.scala:1270)
	at kafka.server.KafkaConfig.<init>(KafkaConfig.scala:1273)
	at kafka.server.KafkaConfig$.fromProps(KafkaConfig.scala:1221)
	at kafka.server.KafkaServerStartable$.fromProps(KafkaServerStartable.scala:34)
	at kafka.server.KafkaServerStartable$.fromProps(KafkaServerStartable.scala:29)
	at kafka.Kafka$.main(Kafka.scala:68)
	at kafka.Kafka.main(Kafka.scala)
```

**Expected behavior**

During initial reading of the documentation, it would have been clear that we should use `extraEnvVars` for adding minor configuration changes to kafka rather than `config`.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
Client: &version.Version{SemVer:"v2.15.1", GitCommit:"cf1de4f8ba70eded310918a8af3a96bfe8e7683b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.15.1", GitCommit:"cf1de4f8ba70eded310918a8af3a96bfe8e7683b", GitTreeState:"clean"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.6-beta.0", GitCommit:"e7f962ba86f4ce7033828210ca3556393c377bcc", GitTreeState:"clean", BuildDate:"2020-01-15T08:26:26Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.9", GitCommit:"2e808b7cb054ee242b68e62455323aa783991f03", GitTreeState:"clean", BuildDate:"2020-01-18T23:24:23Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**

Just to reiterate, I think a docs update to set expectations more clearly is a reasonable fix for this. Happy to submit a small PR if maintainers agree.

**Comment by JoshKCarroll** [comment]
I do see this is sort of mentioned in the docs, so maybe the suggested change is just being more explicit with Config that it needs to specify the entire config when used.

https://github.com/bitnami/charts/tree/master/bitnami/kafka#setting-custom-parameters

**Comment by rafariossaa** [comment]
Hi,
  I think that is included as @JoshKCarroll  indicated. What is your suggestion for the changes in the REAMDE ?, Please don't hesitate to send a PR and we can discuss and included whatever is needed.

**Comment by JoshKCarroll** [comment]
Thanks @rafariossaa you can find my PR here. Happy to discuss further.

https://github.com/bitnami/charts/pull/3971

We spent several hours being confused about this earlier this week, so hoping to help any other users of the chart avoid the same fate (especially as I expect it will get a lot more new usage soon with the deprecation of the `incubator/kafka` chart). I'm not sure about the best practices for docs in this repo though, so if this doesn't meet them let me know. Thank you!

**Comment by zbeedatm** [comment]
After missing with the config of kafka helm chart I can't get it up back! It's getting into CrashLoopBackOff and when looking at the log I can see:

> ERROR Exiting Kafka due to fatal exception (kafka.Kafka$)
org.apache.kafka.common.config.ConfigException: Missing required configuration `zookeeper.connect` which has no default value.
        at kafka.server.KafkaConfig.validateValues(KafkaConfig.scala:1901)
        at kafka.server.KafkaConfig.<init>(KafkaConfig.scala:1896)
        at kafka.server.KafkaConfig.<init>(KafkaConfig.scala:1389)
        at kafka.server.KafkaConfig$.fromProps(KafkaConfig.scala:1327)
        at kafka.Kafka$.buildServer(Kafka.scala:67)
        at kafka.Kafka$.main(Kafka.scala:87)
        at kafka.Kafka.main(Kafka.scala)

How to restore all environment variables?

**Comment by rafariossaa** [comment]
Hi,
  I am glad you were able to solve it.
  Thanks for sharing.
